src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -147,6 +147,10 @@ MergeTreeData::MergeTreeData(
     if (relative_data_path.empty())
         throw Exception("MergeTree storages require data path", ErrorCodes::INCORRECT_FILE_NAME);
 
+    /// Check sanity of MergeTreeSettings. Only when table is created.
+    if (!attach)
```

Comment:
But what about ALTER SETTINGS? https://github.com/clickhouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreeData.cpp#L1653-L1655

Reply:
Yes, we should add sanity check there also...